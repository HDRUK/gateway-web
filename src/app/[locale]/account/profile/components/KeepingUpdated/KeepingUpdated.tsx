import { Fragment } from "react";
import { Control, FieldValues } from "react-hook-form";
import { Divider } from "@mui/material";
import { ComponentTypes } from "@/interfaces/ComponentTypes";
import Box from "@/components/Box";
import InputWrapper from "@/components/InputWrapper";
import Typography from "@/components/Typography";

interface KeepingUpdatedProps<T extends FieldValues> {
    control: Control<T>;
    fields: {
        title: string;
        label: string;
        name: string;
        component: ComponentTypes;
    }[];
}

function KeepingUpdated<T extends FieldValues>({
    control,
    fields,
}: KeepingUpdatedProps<T>) {
    return (
        <Box sx={{ padding: 0 }}>
            <Divider />
            <Typography
                sx={{
                    fontWeight: "bold",
                    paddingTop: 2,
                    paddingBottom: 2,
                }}>
                Keeping you updated
            </Typography>
            {fields.map(field => (
                <Fragment key={field.name}>
                    <Divider />
                    <InputWrapper control={control} {...field} />
                </Fragment>
            ))}

            <Divider />
            <Typography sx={{ marginTop: 2, marginBottom: 4 }}>
                As a user of the Gateway we take the privacy and security of
                your personal data seriously. Our privacy policy aims to give
                you information on how Health Data Research UK collects and
                processes your personal data through your use of this Gateway,
                including any data you may provide by emailing us.
            </Typography>
        </Box>
    );
}

export default KeepingUpdated;
