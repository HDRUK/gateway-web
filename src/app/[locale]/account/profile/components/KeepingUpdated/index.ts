import KeepingUpdated from "./KeepingUpdated";

export default KeepingUpdated;
