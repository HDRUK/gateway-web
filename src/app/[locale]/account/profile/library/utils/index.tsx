import { IconButton } from "@mui/material";
import { createColumnHelper } from "@tanstack/react-table";
import { LibraryListItem, SelectedLibrary } from "@/interfaces/Library";
import Link from "@/components/Link";
import StyledCheckbox from "@/components/StyledCheckbox";
import TooltipIcon from "@/components/TooltipIcon";
import { CheckIcon, DeleteForeverIcon } from "@/consts/icons";
import { RouteName } from "@/consts/routeName";

const columnHelper = createColumnHelper<LibraryListItem>();

const getColumns = ({
    handleSelect,
    handleRemove,
    selected,
    translations,
}: {
    handleSelect: (data: SelectedLibrary) => void;
    handleRemove: (id: string | number) => void;
    selected: SelectedLibrary;
    translations: { [id: string]: string };
}) => [
    columnHelper.display({
        id: "actions",
        meta: { isPinned: true },
        cell: ({ row }) => {
            const {
                id,
                datasetId,
                name,
                dataCustodianId,
                dataCustodian,
                darEnabled,
                darTemplatePublished,
                cohortEnabled,
            } = row.original;
            return (
                <div style={{ textAlign: "center" }}>
                    <StyledCheckbox
                        checked={selected[id]?.selected}
                        onChange={(_e, value) =>
                            handleSelect({
                                [id]: {
                                    selected: value,
                                    datasetId,
                                    name,
                                    teamId: dataCustodianId,
                                    teamName: dataCustodian,
                                    darEnabled,
                                    darTemplatePublished,
                                    cohortEnabled,
                                },
                            })
                        }
                        size="large"
                        sx={{ p: 0 }}
                        iconSx={{ mr: 0 }}
                    />
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),

    columnHelper.display({
        id: "name",
        cell: ({ row: { original } }) => {
            const { datasetId, name } = original;
            const linkHref = `/${RouteName.DATASET_ITEM}/${datasetId}`;

            return (
                <Link href={linkHref}>
                    <div style={{ textAlign: "center" }}>{name}</div>
                </Link>
            );
        },
        header: () => <span>{translations.name}</span>,
        size: 150,
    }),

    columnHelper.display({
        id: "darEnabled",
        cell: ({
            row: {
                original: { darEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {darEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => (
            <TooltipIcon
                buttonSx={{ p: 0 }}
                size="small"
                label={translations.darEnabled}
                content={translations.darEnabled}
            />
        ),
        size: 100,
    }),

    columnHelper.display({
        id: "cohortEnabled",
        cell: ({
            row: {
                original: { cohortEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {cohortEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => <span>{translations.cohortEnabled}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "dataCustodian",
        cell: ({
            row: {
                original: { dataCustodian },
            },
        }) => <div style={{ textAlign: "center" }}>{dataCustodian}</div>,
        header: () => <span>{translations.dataCustodian}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "entityType",
        cell: ({
            row: {
                original: { entityType },
            },
        }) => <div style={{ textAlign: "center" }}>{entityType}</div>,
        header: () => <span>{translations.entityType}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "delete",
        cell: ({ row: { original } }) => {
            return (
                <div style={{ textAlign: "center" }}>
                    <IconButton onClick={() => handleRemove(original.id)}>
                        <DeleteForeverIcon color="primary" />
                    </IconButton>
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),
];

export { getColumns };
