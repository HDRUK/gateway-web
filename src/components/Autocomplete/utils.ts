import { ValueType } from "./Autocomplete";

const getChipLabel = (
    options: { value: string | number; label: string }[],
    value: ValueType
) => {
    if (typeof value === "string") {
        return value;
    }

    return options.find(option => option.value === value)?.label;
};

export { getChipLabel };
