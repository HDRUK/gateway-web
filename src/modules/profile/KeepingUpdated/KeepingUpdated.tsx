import Box from "@/components/Box";
import CheckboxRow from "@/components/CheckboxRow/CheckboxRow";
import { profileContactFormFields } from "@/config/forms/profile";
import { Divider, Typography } from "@mui/material";
import { Fragment } from "react";

interface KeepingUpdatedProps {
    control: string;
}

const KeepingUpdated = ({ control }: KeepingUpdatedProps) => {
    return (
        <Box sx={{ padding: 0 }}>
            <Divider />
            <Typography
                sx={{
                    fontWeight: "bold",
                    paddingTop: 2,
                    paddingBottom: 2,
                }}>
                Keeping you updated
            </Typography>
            {profileContactFormFields.map(field => (
                <Fragment key={field.name}>
                    <Divider />
                    <CheckboxRow control={control} {...field} />
                </Fragment>
            ))}

            <Divider />
            <Typography sx={{ marginTop: 2, marginBottom: 4 }}>
                As a user of the Gateway we take the privacy and security of
                your personal data seriously. Our privacy policy aims to give
                you information on how Health Data Research UK collects and
                processes your personal data through your use of this Gateway,
                including any data you may provide by emailing us.
            </Typography>
        </Box>
    );
};

export default KeepingUpdated;
