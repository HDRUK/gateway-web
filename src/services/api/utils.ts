import { NotificationOptions } from "@/interfaces/Api";
import { AxiosResponse } from "axios";
import { Error } from "@/interfaces/Error";
import notificationService from "../notification";

interface ErrorNotificationProps {
    props: NotificationOptions;
    errorResponse: AxiosResponse<Error>;
    method: "delete" | "post" | "put" | "get";
}

const errorNotification = ({
    errorResponse,
    method,
    props,
}: ErrorNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const { data, status } = errorResponse || {};

    const fallbackTitle = i18n.exists(`api:common.error.status.${status}`)
        ? t(`api:common.error.status.${status}`)
        : "There has been an error";

    const title = data?.title || fallbackTitle;
    const message =
        data?.message ||
        t(`api:common.error.${method}.message`, {
            item: props.itemName || "Item",
        });

    notificationService.apiError(message, {
        title,
        message,
        errors: data?.errors,
        ...notificationProps,
    });
};

interface SuccessNotificationProps {
    props: NotificationOptions;
    method: "delete" | "post" | "put";
}

const successNotification = ({ props, method }: SuccessNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const customMessage = `api:${props.localeKey}.success.${method}.message`;
    const shouldOverideMessage = i18n.exists(customMessage);

    const message = shouldOverideMessage
        ? t(customMessage)
        : t(`api:common.success.${method}.message`, {
              item: props.itemName || "Item",
          });

    notificationService.success(message, { ...notificationProps });
};

export { errorNotification, successNotification };
