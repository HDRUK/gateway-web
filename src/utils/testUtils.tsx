import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
