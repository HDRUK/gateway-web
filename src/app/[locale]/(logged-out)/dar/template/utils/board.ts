import { BoardSections, Board, Task } from "../components";
import { BOARD_SECTIONS } from "../constants";
import { getTasksByStatus } from "./tasks";

export const initializeBoard = (tasks: Task[]) => {
    const boardSections: BoardSections = {};

    Object.keys(BOARD_SECTIONS).forEach(boardSectionKey => {
        boardSections[boardSectionKey] = getTasksByStatus(
            tasks,
            boardSectionKey as Board
        );
    });

    return boardSections;
};

export const findBoardSectionContainer = (
    boardSections: BoardSections,
    id: string
) => {
    if (id in boardSections) {
        return id;
    }

    const container = Object.keys(boardSections).find(key =>
        boardSections[key].find(item => item.id === id)
    );
    return container;
};
