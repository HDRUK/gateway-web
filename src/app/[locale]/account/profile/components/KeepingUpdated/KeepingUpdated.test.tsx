import KeepingUpdated from "./KeepingUpdated";
import { useForm } from "react-hook-form";
import { render } from "@/utils/testUtils";

describe("KeepingUpdated", () => {
    const Component = () => {
        const { control } = useForm();
        return (
            <KeepingUpdated
                fields={[
                    {
                        title: "Feedback",
                        label: "I am happy to be contacted to share and give feedback on my experience with the Gateway",
                        name: "contact_feedback",
                        component: "CheckboxRow",
                    },
                ]}
                control={control}
            />
        );
    };
    it("should render content", () => {
        const wrapper = render(<Component />);
        expect(wrapper.container).toMatchSnapshot();
    });
});
