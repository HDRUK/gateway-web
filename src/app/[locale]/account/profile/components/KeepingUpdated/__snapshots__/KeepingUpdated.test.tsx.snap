// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeepingUpdated should render content 1`] = `
<div>
  <div
    class="MuiBox-root mui-style-old1by"
  >
    <hr
      class="MuiDivider-root MuiDivider-fullWidth mui-style-9mgopn-MuiDivider-root"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 mui-style-13roe7n-MuiTypography-root"
    >
      Keeping you updated
    </p>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth mui-style-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root mui-style-106jpu6"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 mui-style-a793d6-MuiTypography-root"
      >
        Feedback
      </p>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth mui-style-2dhluk-MuiFormControl-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd mui-style-1mbnhdb-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeMedium mui-style-1y1d3h6-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input mui-style-1m9pwf3"
              data-indeterminate="false"
              id="contact_feedback"
              name="contact_feedback"
              type="checkbox"
            />
            <span
              class="mui-style-f6vn6k"
              font-size="medium"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label mui-style-1mg2mqv-MuiTypography-root"
          >
            I am happy to be contacted to share and give feedback on my experience with the Gateway
          </span>
        </label>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth mui-style-9mgopn-MuiDivider-root"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 mui-style-2i2h27-MuiTypography-root"
    >
      As a user of the Gateway we take the privacy and security of your personal data seriously. Our privacy policy aims to give you information on how Health Data Research UK collects and processes your personal data through your use of this Gateway, including any data you may provide by emailing us.
    </p>
  </div>
</div>
`;
